core/app/models/spree/order.rb

```diff
@@ -190,6 +190,12 @@ def self.not_canceled
       where.not(state: 'canceled')
     end
 
+    # => Get all nonzero (reference to :nonzero scope of Spree::Adjustment) adjustments for # => an instance of Spree::Order and its LineItem and Shipment associations
+    def absolute_adjustments
+      Spree::Adjustment.includes(:order).where(order: self, adjustable_type: ['Spree::LineItem', 'Spree::Shipment'])
+      .where.not(amount: 0).or(Spree::Adjustment.includes(:order).where(adjustable: self).where.not(amount: 0))
+    end
```

Comment:
I believe that the we can do this without using `or` by relying on the `all_adjustments` association on the order. We can also use the existing scope that you reference in the comment. Isn't this equivalent? In what situations is it not?

```suggestion
    def absolute_adjustments
      all_adjustments.includes(:order).nonzero
    end
```

Reply:
If you look closely the all_adjustments association of `Spree::Orders` doesn't contemplate adjustments that have the `Spree::LineItem` and `Spree::Shipment` associations of `Spree::Order` as their adjustable target model. These different adjustments are called at `<% if order.adjustments.nonzero.exists? || order.line_item_adjustments.nonzero.exists? || order.shipment_adjustments.nonzero.exists? || order.shipments.any? %>` 

What `.or` does in this circunstance is that it replaces the logical union of the 3 different `Exists?` queries and makes just one of them on the database rather than in Ruby. Before `.or` you have all adjustments to `LineItem` and `Shipment` that have an `order_id` of the current order. But that order is not the adjustable object of the query.

Reply:
```ruby
# => Get all nonzero (reference to :nonzero scope of Spree::Adjustment) adjustments for an instance of Spree::Order and its LineItem and Shipment associations
def absolute_adjustments
  all_adjustments.nonzero.or(Spree::Adjustment.where(adjustable: order).nonzero)
end
```

I thought of something that accomplishes a similar thing with just 1 call rather than 3 as it was before. I'm pushing to see test results.